platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/DefaultRootTestEventReporter.java

```diff
@@ -69,9 +69,18 @@ class DefaultRootTestEventReporter extends DefaultGroupTestEventReporter {
 
     @Override
     public void close() {
-        super.close();
+        try {
+            super.close();
+        } catch (Throwable t) {
+            // Ensure binary results are written to disk.
+            try {
+                testResultWriter.close();
+            } catch (IOException e) {
+                t.addSuppressed(e);
```

Comment:
🤔 This seems potentially confusing.  The exception `t` was caused prior to the exception `e`.  So why are we reporting `e` as a "suppressed" exception of `t`?  I don't think this properly meets the criteria of "sibling exceptions" this was made for.

It seems more like we should throw a new "Multiple exceptions occurred closing the reporter" exception, which includes both `t` and `e` as causes/suppressed exceptions.

Reply:
I think this makes sense, in a try-with-resources statement the original throwable `t` is what is thrown, and the exceptions from incidental clean-up are added to `t` as suppressed exceptions. AFAICT this is identical to something like:
```
try (var temp = testResultWriter) {
  super.close();
}
```
so I think it's acceptable.